### ISSUE TITLE
[BUG] File creation in Solution explorer doesn't delete

### ISSUE THREAD

**Issue by joeizang** [microsoft/vscode-dotnettools#280]
### Describe the Issue

if the filename to be created has a backslash just before the extension, then a folder gets created with with the name of the intended for the file and the file itself ends up being a ".<fileextension>" Both file and folder cannot be deleted in solution explorer. It only shows an error indicating it cannot delete the file or folder. This happened to me when I accidentally created a file with the backslash so I ended up with "CsharpCodefile\.cs".

### Steps To Reproduce

1. Right-click on a folder and create a class or any other code file.
2. When typing the name of the file to be created, add a "\" backslash just before your extension so like this "CodeFile\.cs"
3. A file gets opened in the editor but in the solution explorer, a folder gets created and a .cs file will live in the folder.
4. Try to delete the file or folder and an error will result.

### Expected Behavior

There should be a list of what are considered illegal characters that are not allowed in file names, so if such a character is being used, at least the file or folder should not be created. Rather an either a modal or alert should indicate that the operation should be repeated again or an empty prompt should result asking for the name to be typed again.

### Environment Information

- OS Mac OS Ventura 13.3
- VSCode version 1.80.1
- C# Dev Kit version 0.2.100

**Comment by smitpatel** [comment]
Which version of .NET SDK are you using?

When I tried to follow repro steps, step 3 didn't occur. No file was created. Looking at logs this is what mentioned
```
Calling dotnet new failed with error: ERROR ExitCode(127)
Details
Name cannot contain any of the following characters '"', '<', '>', '|', ':', '*', '?', '\', '/' or character codes char(0), char(1), char(2), char(3), char(4), char(5), char(6), char(7), char(8), char(9), char(10), char(11), char(12), char(13), char(14), char(15), char(16), char(17), char(18), char(19), char(20), char(21), char(22), char(23), char(24), char(25), char(26), char(27), char(28), char(29), char(30), char(31)

For details on the exit code, refer to https://aka.ms/templating-exit-codes#127
```
So dotnet new is doing right thing to block the name. (We may need to bubble up better error for user)

**Comment by joeizang** [comment]
I am using .NET 7 (latest) and .NET 8 preview 6. The projects I got the erroneous behaviour on is .NET 7

**Comment by smitpatel** [comment]
If you have SDK below 7.0.200 then we would create file without dotnet new for class which may show above behavior.

cc: @kvenkatrajan - thoughts?

**Comment by kvenkatrajan** [comment]
We have discussed and are investigating the issue.

**Comment by smitpatel** [comment]
We will throw error message for invalid character in file name when above error is encountered.

**Comment by joeizang** [comment]
@smitpatel I am assuming that invalid characters will consider all OSes?

**Comment by smitpatel** [comment]
We defer to the templating code for that. We will just intercept and bubble up message from "dotnet new" when there are invalid characters.

**Comment by joeizang** [comment]
> We defer to the templating code for that. We will just intercept and bubble up message from "dotnet new" when there are invalid characters.

But when creating a solution folder and I use invalid characters (which was how I got my error to begin with), how would "dotnet new" help with invalid characters? shouldn't this be handled somewhere else higher up?

**Comment by smitpatel** [comment]
The original repro steps, mentions about using backslash when creating new file. Creation of new file in C# dev kit happens through dotnet new CLI. So "dotnet new" covers all invalid characters in file creation scenario.
Are you having issue with creating solution folder with invalid characters? (That would be different issue since root cause/fix is different)

**Comment by joeizang** [comment]
> The original repro steps, mentions about using backslash when creating new file. Creation of new file in C# dev kit happens through dotnet new CLI. So "dotnet new" covers all invalid characters in file creation scenario. Are you having issue with creating solution folder with invalid characters? (That would be different issue since root cause/fix is different)

no what you described is exactly my case, just I didn't realize that file creation in the C# dev kit uses the cli to do this. (this actually makes sense). Lets hope that the cli sorts this out.

**Comment by smitpatel** [comment]
Fixed in C# dev kit prerelease version 0.5.2
